contracts/patriciaTree/Data.sol

```diff
@@ -45,7 +45,8 @@ library Data {
     if (pos == 0) {
       prefix.data = bytes32(0);
     } else {
-      prefix.data = bytes32(uint(self.data) & ~uint(1) << 255 - pos); // ignore-swc-101
+      // prefix.data = bytes32(uint(self.data) & ~uint(1) << 255 - pos); // ignore-swc-101
+      prefix.data = self.data & ~bytes32((uint(1) << (256 - pos)) - 1);
```

Comment:
I reviewed this... apart from `pos == 256` which seems to return a wrong result (a mask of `00...00` instead of the expected `11...11` due to the underflow, the other values I checked (`pos == 1`, `pos == 255`) returned the same values.

Reply:
Noting that in a call with @area we determined that this function was only ever called with a value of `pos = 255 - k` and so the buggy input of `256` was never seen.